### ISSUE TITLE
MacOS precompiled binary - build or distribution issue

### ISSUE THREAD

**Issue by thomoco** [ClickHouse/ClickHouse#38648]
> You have to provide the following information whenever possible.

I think we may have some sort of build or download problem here with our MacOS builds. Seeing some very weird behavior on the x86_64 vs. ARM precompiled binaries for MacOS.

I have a Macbook Pro M1 (Apple Silicon) so would think the AArch64 binary should be the correct one, but it is not able to start and exiting with an error code, whereas the x86_64 binary does run.

Here is the MacBook Pro chipset info:
```
% sysctl -a | egrep -i 'machdep.cpu|kern.version'
kern.version: Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:29 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T8101
machdep.cpu.cores_per_package: 8
machdep.cpu.core_count: 8
machdep.cpu.logical_per_package: 8
machdep.cpu.thread_count: 8
machdep.cpu.brand_string: Apple M1
```

> A clear and concise description of what works not as it is supposed to.

Yes, using the latest release, as downloaded via the curl command provided on the installation page: https://clickhouse.com/docs/en/getting-started/install/
```
% ./clickhouse server
2022.06.30 10:51:51.496780 [ 4703870 ] {} <Information> : Starting ClickHouse 22.7.1.1011 with revision 54464, no build id, PID 12225
```

Oddly, the x86_64 download will run on this Macbook Pro M1, unexpectedly:
```
% curl -O 'https://builds.clickhouse.com/master/macos/clickhouse' && chmod a+x ./clickhouse
% ./clickhouse server
...
2022.06.30 10:53:05.712263 [ 4705008 ] {} <Information> Application: Ready for connections.
```

I was expecting that only the aarch64 (Apple Silicon) build should be the right precompiled binary. However, it does not and seems to die with a signal 9 and exit status code of 137, which appears to possibly be an OOM killer:
```
% curl -O 'https://builds.clickhouse.com/master/macos-aarch64/clickhouse' && chmod a+x ./clickhouse
$ ./clickhouse server
Killed: 9
$ echo $?
137
```

The file seems to be an Arm64 build though:
```
$ file clickhouse
clickhouse: Mach-O 64-bit executable arm64
```

> If possible, change "enabled" to true in "send_crash_reports" section in `config.xml`:

did not do this yet, can do as followup

**How to reproduce**

1. Download AArch64 version
```
$ curl -O 'https://builds.clickhouse.com/master/macos-aarch64/clickhouse' && chmod a+x ./clickhouse
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  328M  100  328M    0     0  2987k      0  0:01:52  0:01:52 --:--:-- 3943k
```
2. Execute binary
```
$ ./clickhouse help
Killed: 9
```

**Expected behavior**
Expected AArch64 build to run on this Macbook Pro, but oddly only the x86_64 binary runs.

> A clear and concise description of what you expected to happen.

**Error message and/or stacktrace**

> If applicable, add screenshots to help explain your problem.

**Additional context**

> Add any other context about the problem here.

**Comment by den-crane** [comment]
https://github.com/Altinity/homebrew-clickhouse

**Comment by thomoco** [comment]
thank you @den-crane, although we will need to fix the official ClickHouse builds

**Comment by rschu1ze** [comment]
The x86 Mac binary (--> https://clickhouse.com/docs/en/getting-started/install/) works on M1/M2 because of Rosetta.

Interestingly, the ARM binary installed from the same source works just fine on my Apple M1.

Would be interesting to hear from other Mac users.

**Comment by thomoco** [comment]
I just retested again, after doing a reboot and making sure I had no data preloaded (clean-room test), and the AArch64 binary is working on my M1 also.

However, when I test the same AArch64 binary from the prior directory where I did have significant clickhouse data preloaded locally on disk, I can still reproduce the problem. So I'm assuming now it really was an OOM as killed by the kernel or shell due to memory usage (running as a non-privileged user).

Interesting though that the x86_64 version (with Rosetta) executes ok with the preloaded data, but not the AArch64 version.

In any case, I'll close this issue as the binary itself seems fine. Thanks for looking into this.
